src/lib.rs

```diff
@@ -197,62 +183,144 @@ where
+            Some(info.data)
+        } else { None }
+    }
+
+    pub fn get_finalized_head(&self) -> WsResult<String> {
+        Self::_get_request(self.url.clone(), json_req::chain_get_finalized_head().to_string())
+    }
+
+    pub fn get_header(&self, hash: Option<Hash>) -> WsResult<String> {
+        Self::_get_request(self.url.clone(), json_req::chain_get_header(hash).to_string())
+    }
 
-        hexstr_to_account_data(result_str).ok()
+    pub fn get_block(&self, hash: Option<Hash>) -> WsResult<String> {
+        Self::_get_request(self.url.clone(), json_req::chain_get_block(hash).to_string())
     }
 
     pub fn get_request(&self, jsonreq: String) -> WsResult<String> {
         Self::_get_request(self.url.clone(), jsonreq)
     }
 
-    pub fn get_storage(
+    pub fn get_storage_value<V:Decode+Clone>(
         &self,
-        storage_prefix: &str,
-        storage_key_name: &str,
-        param: Option<Vec<u8>>,
-    ) -> WsResult<String> {
-        Self::_get_storage(self.url.clone(), storage_prefix, storage_key_name, param)
+        storage_prefix: &'static str,
+        storage_key_name: &'static str,
+    ) -> Option<V> {
+        let storagekey: sp_core::storage::StorageKey = self.metadata
+                .module(storage_prefix).unwrap()
+                .storage(storage_key_name).unwrap()
+                .get_value().unwrap()
+                .key();
+        info!("storage key is: 0x{}", hex::encode(storagekey.0.clone()));
+        self.get_storage_by_key_hash(storagekey.0)
     }
 
-    pub fn get_storage_double_map(
+    pub fn get_storage_map<K:Encode, V:Decode+Clone>(
         &self,
-        storage_prefix: &str,
-        storage_key_name: &str,
-        first: Vec<u8>,
-        second: Vec<u8>,
-    ) -> WsResult<String> {
-        Self::_get_storage_double_map(
-            self.url.clone(),
-            storage_prefix,
-            storage_key_name,
-            first,
-            second,
-        )
+        storage_prefix: &'static str,
+        storage_key_name: &'static str,
+        map_key: K,
+    ) -> Option<V> {
+        let storagekey: sp_core::storage::StorageKey = self.metadata
+                .module(storage_prefix).unwrap()
+                .storage(storage_key_name).unwrap()
+                .get_map::<K, V>().unwrap()
+                .key(map_key);
+        info!("storage key is: 0x{}", hex::encode(storagekey.0.clone()));
+        self.get_storage_by_key_hash(storagekey.0)
     }
 
-    pub fn send_extrinsic(&self, xthex_prefixed: String) -> WsResult<Hash> {
+    pub fn get_storage_by_key_hash<V:Decode+Clone>(&self, hash: Vec<u8>) -> Option<V> {
+        let mut keyhash_str = hex::encode(hash);
+        keyhash_str.insert_str(0, "0x");
+        let jsonreq = json_req::state_get_storage(&keyhash_str);
+        if let Ok(hexstr) = Self::_get_request(self.url.clone(), jsonreq.to_string()) {
+            info!("storage hex = {}", hexstr);
+            let hexstr = hexstr
```

Comment:
Maybe we should generify the `utils::hexstr_vec` function to be able to call it here?

`pub fn decode_from_hexstr<V: Decode>(hexstr: String) -> Option<V>/Result<V, FromHexError>`